common/lib/xmodule/xmodule/modulestore/imscc_exporter.py

```diff
@@ -0,0 +1,774 @@
+        return self.modulestore.get_course(courselike_key, depth=None, lazy=False)
+
+    def get_sequential_modules(self, modulestore, courselike_key):
+        """
+        Retrieve all sequential modules from the course
+        """
+        with modulestore.branch_setting(ModuleStoreEnum.Branch.published_only, courselike_key):
+            # Get all top-level modules (e.g., chapters, sequentials)
+            top_level_modules = modulestore.get_items(courselike_key)
+
+            sequentials = []
+            for module in top_level_modules:
+                if module.category == 'sequential':
+                    sequentials.append(module)
+        return sequentials
+
+    def get_chapter_modules(self, modulestore, courselike_key):
+        """
+        Retrieve all chapter modules from the course
+        """
+        with modulestore.branch_setting(ModuleStoreEnum.Branch.published_only, courselike_key):
+            # Get all top-level modules (e.g., chapters, sequentials)
+            top_level_modules = modulestore.get_items(courselike_key)
+
+            chapter = []
+            for module in top_level_modules:
+                if module.category == 'chapter':
+                    chapter.append(module)
+        return chapter
+
+    def get_chapter_sequential_modules(self, modulestore, courselike_key):
+        """
+        Retrieve all chapter and sequential modules from the course
+        """
+        with modulestore.branch_setting(ModuleStoreEnum.Branch.published_only, courselike_key):
+            # Get all top-level modules (e.g., chapters, sequentials)
+            top_level_modules = modulestore.get_items(courselike_key)
+
+            sequentials_chapters = []
+            for module in top_level_modules:
+                if module.category == 'sequential' or module.category == 'chapter':
+                    sequentials_chapters.append(module)
+        return sequentials_chapters
+
+    def serialize_chapter_sequential(self, chapter_sequential):
+        """
+        Return a serialized object of an inputted chpater or sequential in order to bypass dictionary key issues
+        """
+        return SerializableChapterSequential(
+        chapter_sequential.display_name,
+        chapter_sequential.format,
+        chapter_sequential.url_name,
+        chapter_sequential.category,
+        chapter_sequential.course_id
+    )
+
+    def get_course_abbreviation(self, courselike_key):
+        """
+        Returns a course abbreviation to append to the start of module and assignment names
+        Returns nothing if the courselike_key provided doesn't match the expected pattern in most courses
+        """
+        # re pattern for extracting the values between the plus signs
+        between_pluses = r'(?<=\+)(.*?)(?=\+)'
+        # re pattern for the 'FAA-ACS-AM-IA-ACE' that all courses seem to have
+        course_type = r'([A-Za-z]{3}-[A-Za-z]{3}-[A-Za-z]{2}-[A-Za-z]{2}-[A-Za-z]{3})'
```

Comment:
@mystica-l 
This only works for things like `FAA-ACS-AM-IB-ACD` for example. It will not work for `EV-ST-IEV` or other things like `SFT108`. I'm going to remove this check and we'll just use the full Course Number for now. At the moment, there is no consistency in how we name these values.